src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -147,27 +149,37 @@ public CacheTagHelper(IMemoryCache memoryCache, HtmlEncoder htmlEncoder) : base(
         // Internal for unit testing
         internal MemoryCacheEntryOptions GetMemoryCacheEntryOptions()
         {
+            var hasEvictionCriteria = false;
             var options = new MemoryCacheEntryOptions();
             if (ExpiresOn != null)
             {
+                hasEvictionCriteria = true;
                 options.SetAbsoluteExpiration(ExpiresOn.Value);
             }
 
             if (ExpiresAfter != null)
             {
+                hasEvictionCriteria = true;
                 options.SetAbsoluteExpiration(ExpiresAfter.Value);
             }
 
             if (ExpiresSliding != null)
             {
+                hasEvictionCriteria = true;
                 options.SetSlidingExpiration(ExpiresSliding.Value);
             }
 
             if (Priority != null)
             {
+                hasEvictionCriteria = true;
                 options.SetPriority(Priority.Value);
             }
 
+            if (!hasEvictionCriteria)
```

Comment:
Should this be baked into MemoryCache? Something along the lines of 
```
DefaultEvictionPolicy
{
   Absolute, Sliding, None
}
```

and consumers opt out if they explicitly want long-lived caches?